src/components/MyAccount/Settings/AccountSettingsDisplayOptions.tsx

```diff
@@ -39,35 +40,57 @@ export const AccountSettingsForm = ({ patron }: { patron: Patron }) => {
         | Iterable<ReactNode>
       switch (setting.term) {
         case "Home library":
-          inputField = (
-            <Select
-              name={setting.field}
-              id="update-home-library-selector"
-              labelText="Update home library"
-              showLabel={false}
-            >
-              <option value={patron.homeLibrary.code}>
-                {patron.homeLibrary.name}
-              </option>
-            </Select>
-          )
+          {
+            const patronHomeLibrary = getLibraryByCode(patron.homeLibraryCode)
+            const sortedPickupLocations = [
+              patronHomeLibrary,
+              ...filteredPickupLocations.filter(
+                (loc) => loc.code.trim() !== patron.homeLibraryCode.trim()
+              ),
+            ]
+            inputField = (
+              <Select
+                name={setting.field}
+                id="update-home-library-selector"
+                labelText="Update home library"
+                showLabel={false}
+              >
+                {sortedPickupLocations.map((loc, i) => (
+                  <option key={`location-option-${i}`} value={loc.code}>
+                    {loc.name}
+                  </option>
+                ))}
+              </Select>
+            )
+          }
           break
-        case "Notification preference":
+        case "Notification preference": {
+          const patronNotPref =
+            notificationPreferenceTuples.find(
+              (pref) => pref[1] === patron.notificationPreference
+            ) || notificationPreferenceTuples[0]
+          const sortedNotPrefs = [
+            patronNotPref,
+            ...notificationPreferenceTuples.filter(
+              (pref) => pref[1] !== patronNotPref[1]
+            ),
+          ]
           inputField = (
             <Select
               name={setting.field}
               id="notification-preference-selector"
               labelText="Update notification preference"
               showLabel={false}
             >
-              {Object.keys(notificationPreferenceMap).map((pref) => (
-                <option key={pref + "-option"} value={pref}>
-                  {notificationPreferenceMap[pref]}
+              {sortedNotPrefs.map((pref) => (
+                <option key={pref + "-option"} value={pref[0]}>
+                  {pref[1]}
                 </option>
               ))}
             </Select>
           )
           break
+        }
```

Comment:
The braces is a bit confusing and my editor also confuses the syntax. This seems fine but maybe move the `break` outside of the curly brace similar to how the "Home library" case is written.